docs/addons/karpenter.md

```diff
@@ -0,0 +1,124 @@
+# Karpenter Add-on
+
+Karpenter add-on is based on the [Karpenter](https://github.com/aws/karpenter) open source node provisioning project. It provides a more efficient and cost-effective way to manage workloads by launching just the right compute resources to handle a cluster's application. 
+
+Karpenter works by:
+
+* Watching for pods that the Kubernetes scheduler has marked as unschedulable,
+* Evaluating scheduling constraints (resource requests, nodeselectors, affinities, tolerations, and topology spread constraints) requested by the pods,
+* Provisioning nodes that meet the requirements of the pods,
+* Scheduling the pods to run on the new nodes, and
+* Removing the nodes when the nodes are no longer needed
+
+## Prerequisite
+
+Here are the prerequisites:
+1. Karpenter discovers subnets tagged `kubernetes.io/cluster/$CLUSTER_NAME`. Private subnets where the EKS cluster is deployed shoule be tagged as such.
+2. (If using Spot), EC2 Spot Service Linked Role should be created. See [here](https://docs.aws.amazon.com/batch/latest/userguide/spot_fleet_IAM_role.html) for more details.
+
+## Usage
+
+```typescript
+import * as ssp from '@aws-quickstart/ssp-amazon-eks';
+
+const addOn = new ssp.addons.KarpenterAddOn();
+const addOns: Array<ClusterAddOn> = [ addOn ];
+
+const app = new cdk.App();
+new EksBlueprint(app, 'my-stack-name', addOns, [], {
+  env: {
+      account: <AWS_ACCOUNT_ID>,
+      region: <AWS_REGION>,
+  },
+});
+```
+
+To validate that Karpenter add-on is running ensure that the add-on deployments for the controller and the webhook are in `RUNNING` state:
+
+```bash
+# Assuming add-on is installed in the karpenter namespace.
+$ kubectl get po -n karpenter
+NAME                                    READY   STATUS    RESTARTS   AGE
+karpenter-controller-79dc99d7cd-zkkf7   1/1     Running   0          62m
+karpenter-webhook-7bf684c676-52chv      1/1     Running   0          62m
+```
+
+## Functionality
+
+1. Creates Karpenter Node Role, Karpenter Instance Profile, and Karpenter Controller Policy.
+2. Creates `karpenter` namespace.
+3. Creates Kubernetes Service Account, and associate AWS IAM Role with Karpenter Controller Policy attached using [IRSA](https://docs.aws.amazon.com/emr/latest/EMR-on-EKS-DevelopmentGuide/setting-up-enable-IAM.html).
+4. Deploys Karpenter helm chart in the `karpenter` namespace, configuring cluster name and cluster endpoint on the controller by default.
+
+## Using Karpenter
+
+To use Karpenter, you need to provision a Karpenter [provisioner CRD](https://karpenter.sh/docs/provisioner/). A single provisioner is capable of handling many different pod shapes.
+
+Create a default provisioner using the command below:
```

Comment:
Should we create a default provisioner as per [`spec.requirements`](https://karpenter.sh/docs/provisioner/#specrequirements)? We could accept the parameters for each requirement and configure the spec dynamically.

Reply:
Added.